packages/service-worker/src/update.ts

```diff
@@ -23,13 +24,36 @@ import {ERR_SW_NOT_SUPPORTED, NgswCommChannel, UnrecoverableStateEvent, UpdateAc
  */
 @Injectable()
 export class SwUpdate {
+  /**
+   * Emits a `VersionDetectedEvent` event whenever a new version is detected on the server.
+   *
+   * Emits a `VersionInstallationFailedEvent` event whenever checking for or downloading a new
+   * version fails.
+   *
+   * Emits a `VersionReadyEvent` event whenever a new version has been downloaded and is ready for
+   * activation.
+   */
+  readonly versionUpdates: Observable<VersionEvent>;
+
   /**
    * Emits an `UpdateAvailableEvent` event whenever a new app version is available.
+   *
+   * @deprecated Use {@link versionUpdates} instead.
+   *
+   * The of behavior `available` can be rebuild by filtering for the `VersionReadyEvent`:
+   * ```
+   * this.versionUpdates.pipe(
+   *   filter((evt: VersionEvent): evt is VersionReadyEvent => evt.type === 'VERSION_READY'),
+   * ).subscribe(...)
```

Comment:
We should not use `this.versionUpdates` here :grin:
Also, we don't need to show `.subscribe()`, since `SwUpdate#available` has to be subscribed to as wll (i.e. it is not a hot observable). Something like the following would be more accurate imo:

```suggestion
   * import {filter, map} from 'rxjs/operators';
   * // ...
   * const updatesAvailable = swUpdate.versionUpdates.pipe(
   *   filter((evt): evt is VersionReadyEvent => evt.type === 'VERSION_READY'),
   *   map(evt => ({
   *     type: 'UPDATE_AVAILABLE',
   *     current: evt.currentVersion,
   *     available: evt.latestVersion,
   *   })));
```